### ISSUE TITLE
Training with low-resource Indo-Aryan/Dravidian languages

### ISSUE THREAD

**Issue by ronaldtse** [secryst/secryst#16]
https://www.aclweb.org/anthology/2020.acl-srw.22.pdf

**Comment by ronaldtse** [comment]
This is a good example for interscript usage.

Training steps:

1. Transliterate low-resource scripts (edge script, e.g. Gujarati) into a high-resource script (hub script, e.g. Hindi)
2. Train the dataset of the edge language with the hub language (concatenated dataset of the same script)

Translate steps:

1. Transliterate low-resource scripts (edge script, e.g. Gujarati) into a high-resource script (hub script, e.g. Hindi)
2. Translate the edge language into the hub language using the trained model.
3. Transliterate the hub script back to the edge script 

> For all of our experiments, we use the OpenNMT-py (Klein et al., 2018) toolkit . We use the Trans-
former model with 6 layers in both the encoder and decoder, each with 512 hidden units. The word embedding size is set to 512 with 8 heads. The training is done in batches of maximum 4096 to- kens at a time with dropout set to 0.3. We use the Adam (Kingma and Ba, 2014) optimizer to opti- mize model parameters. We validate the model every 5,000 steps via BLEU (Papineni et al., 2002) and perplexity on the development set. We train all our NMT models for 150k steps except for fine- tuning which is done for 10k steps. After transla- tion at the test time, we rejoin the translated BPE segments and convert the translated sentences back to their original language scripts. Finally, we eval- uate the accuracy of our translation models using BLEU.
